library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java

```diff
@@ -96,34 +97,30 @@ private void buildActionIntents(String tag, int type, Collection<DownloadBatch>
             builder.setContentIntent(PendingIntent.getBroadcast(context, 0, clickIntent, PendingIntent.FLAG_UPDATE_CURRENT));
             builder.setOngoing(true);
 
-            DownloadBatch batch = cluster.iterator().next();
             Intent cancelIntent = new Intent(Constants.ACTION_CANCEL, null, context, DownloadReceiver.class);
             cancelIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batch.getBatchId());
             PendingIntent pendingCancelIntent = PendingIntent.getBroadcast(context, 0, cancelIntent, PendingIntent.FLAG_UPDATE_CURRENT);
             builder.addAction(R.drawable.dl__ic_action_cancel, context.getString(R.string.dl__cancel), pendingCancelIntent);
 
-        } else if (type == DownloadNotifier.TYPE_SUCCESS) {
-            DownloadBatch batch = cluster.iterator().next();
+        } else if (type == DownloadNotifier.TYPE_SUCCESS || type == DownloadNotifier.TYPE_CANCELLED) {
             // TODO: Decide how we handle notification clicks
```

Comment:
I think it's related to the `clickIntent` which is now on Line `119`. If there's no extra default behaviour that we want to add here then the comment can probably be removed